src/main/java/ro/utcn/sp1tz3/Assignment1/controller/ConsoleController.java

```diff
@@ -0,0 +1,254 @@
+
+    private void handleAddQuestion(){
+        if(log) {
+            System.out.println("title");
+            scanner.nextLine();
+            String title = scanner.nextLine();
+            System.out.println("text");
+            String text = scanner.nextLine();
+            Date date = new Date();
+            Timestamp ts = new Timestamp(date.getTime());
+            System.out.println("Add Tag(s)");
+            ArrayList<Tag> setTag = new ArrayList<>();
+            List<Tag> tags = tagManagementService.listTags();
+            String exit = "start";
+            while(!exit.equals("exit")) {
+                boolean exists = false;
+                String tag = scanner.next();
+                exit = tag;
+                for(Tag t: tags){
+                    if(t.getTitle().equals(tag)){
+                        setTag.add(t);
+                        exists = true;
+                    }
+                }
+                if(!exists && !tag.equals("exit"))
+                    setTag.add(new Tag(tag));
+            }
+            Question question = questionManagementService.addQuestion(currentUser.getUserid(), title, text, ts, setTag);
+            System.out.println("Added question " + question.toString());
+        } else {
+            System.out.println("You must be logged in.");
+        }
+    }
+
+    private void handleAddAnswer(){
+        if(log) {
+            System.out.println("Question id");
+            int questionid = scanner.nextInt();
+            System.out.println("text");
+            scanner.nextLine();
+            String text = scanner.nextLine();
+            Date date = new Date();
+            Timestamp ts = new Timestamp(date.getTime());
+            Answer answer = answerManagementsService.addAnswer(questionid, currentUser.getUserid(), text, ts);
+            System.out.println("Answered question " + questionid + " with " + answer + ".");
+        } else {
+            System.out.println("You must be logged in.");
+        }
+    }
+
+
+    private void handleRemove(){
+        if(log){
+            System.out.println("Question id");
+            int id = scanner.nextInt();
+            if(currentUser.getUserid().equals(questionManagementService.get(id).getUserid())){
+                questionManagementService.removeQuestion(id);
+                System.out.println("Removed question " + id);
+            } else {
+                System.out.println("You can't delete other peoples questions.");
+            }
+        }
+    }
+
+    private void handleLogoff(){
+        log = false;
+        System.out.println("Logged off.");
+    }
+
+    private void handleListUsers(){
+        userManagementService.listUsers().forEach(s -> System.out.println(s.toString()));
+    }
+
+    private void handleFilterQuestions(){
+        if(log) {
+            System.out.println("Text to filter");
+            String filter = scanner.next().trim();
+            questionManagementService.listQuestions().forEach(s -> {
```

Comment:
You should do this inside the repositories actually, not in the presentation layer.